src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java

```diff
@@ -266,13 +86,50 @@ private static void assertGetRecords(MtAmazonDynamoDbStreams streams, Collection
                 AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
 
             return java.util.stream.Stream.of(
-                Arguments.of(indexMtDynamoDb, indexMtDynamoDbStreams),
-                Arguments.of(indexBinaryHkMtDynamoDb, indexBinaryHkMtDynamoDbStreams),
-                Arguments.of(tableMtDynamoDb, tableMtDynamoDbStreams)
+                Arguments.of(indexMtDynamoDb, indexMtDynamoDbStreams, indexMtDynamoDbStreamsCache),
+                Arguments.of(indexBinaryHkMtDynamoDb, indexBinaryHkMtDynamoDbStreams,
+                    indexBinaryHkMtDynamoDbStreamsCache),
+                Arguments.of(tableMtDynamoDb, tableMtDynamoDbStreams, null)
             );
         }
     }
 
+    /**
+     * Verifies describeStreamCache results match what's expected for a given key.
+     */
+    private void assertDescribeStreamCache(MtAmazonDynamoDbStreams mtDynamoDbStreams,
+                                           CachingAmazonDynamoDbStreams cachingStreams,
+                                           Stream stream,
+                                           boolean expectedCacheHit,
+                                           DescribeStreamResult expectedResult) {
+        // Setup
+        String key = stream.getStreamArn();
+        Cache<String, DescribeStreamResult> describeStreamCache = cachingStreams.getDescribeStreamCache();
+
+        // Verify cache hit/miss based on expected result
+        StreamsTestUtil.verifyDescribeStreamCacheResult(describeStreamCache, key, expectedCacheHit, expectedResult);
+
+        // Trigger flow that interacts with the describe stream cache
+        Optional<String> shardIterator = BASE_TEST_UTILS.getShardIterator(mtDynamoDbStreams, stream);
+        assertTrue(shardIterator.isPresent());
```

Comment:
Why assert that we can get a shard iterator? That seems beyond the scope of this test?

Reply:
Because the `shardIterator` gets used to verify the shard id's match a couple lines below:
https://github.com/salesforce/mt-dynamo/blob/cb16e79927723ba9641d53ff1ea6ca27bfcef740/src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java#L120